File: fix.py
#!/usr/bin/env python3
import re

# Read the file
with open('./src/app/components/ActiveChatView.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

print(f"Original file length: {len(content)}")

# Pattern 1: Replace reasoning steps
pattern1 = r'<div className="space-y-4 text-\[14px\]">\s+\{/\* Step 1 - Understanding Request \*/\}[\s\S]*?\{/\* Step 5 - Building Arguments \*/\}[\s\S]*?</motion\.div>\s+\)\}\s+</div>'

replacement1 = '''<div className="space-y-4 text-[14px]">
                      <DynamicReasoningSteps 
                        reasoningSteps={reasoningSteps}
                        reasoningContent={reasoningContent}
                      />
                    </div>'''

if re.search(pattern1, content):
    content = re.sub(pattern1, replacement1, content)
    print("✅ Replaced reasoning steps")
else:
    print("❌ Could not find reasoning steps pattern")

# Pattern 2: Replace source items  
pattern2 = r'<div className="text-\[14px\]">\s+\{/\* Source Item 1 - Practical Law \*/\}[\s\S]*?\{/\* Source Item 6 - Government Legal Database \*/\}[\s\S]*?</motion\.div>\s+\)\}\s+</div>'

replacement2 = '''<div className="text-[14px]">
                      <DynamicSourceItems 
                        sourcesItems={sourcesItems}
                        sourceContent={sourceContent}
                      />
                    </div>'''

if re.search(pattern2, content):
    content = re.sub(pattern2, replacement2, content)
    print("✅ Replaced source items")
else:
    print("❌ Could not find source items pattern")

# Write the file
with open('./src/app/components/ActiveChatView.tsx', 'w', encoding='utf-8') as f:
    f.write(content)

print(f"New file length: {len(content)}")
print("\n✅ DONE!")

File: fix.js
const fs = require('fs');
const path = require('path');

// Read the file
const filePath = path.join(__dirname, 'src', 'app', 'components', 'ActiveChatView.tsx');
let content = fs.readFileSync(filePath, 'utf8');

// Split into lines for easier manipulation
let lines = content.split('\n');

// FIX 1: Replace reasoning steps (lines ~1650-1742)
let reasoningStart = -1;
let reasoningEnd = -1;

for (let i = 0; i < lines.length; i++) {
  if (lines[i].includes('space-y-4 text-[14px]') && i > 1640 && i < 1660) {
    reasoningStart = i + 1;
    // Find the matching closing div
    let depth = 0;
    for (let j = i + 1; j < Math.min(i + 100, lines.length); j++) {
      if (lines[j].includes('</div>') && lines[j].trim() === '</div>' && j > 1740 && j < 1750) {
        reasoningEnd = j;
        break;
      }
    }
    break;
  }
}

if (reasoningStart > 0 && reasoningEnd > 0) {
  const newReasoning = [
    '                      <DynamicReasoningSteps ',
    '                        reasoningSteps={reasoningSteps}',
    '                        reasoningContent={reasoningContent}',
    '                      />'
  ];
  lines.splice(reasoningStart, reasoningEnd - reasoningStart, ...newReasoning);
  console.log(`✅ Fixed reasoning steps (lines ${reasoningStart+1}-${reasoningEnd+1})`);
}

// FIX 2: Replace source items (lines ~1797-1893)
let sourcesStart = -1;
let sourcesEnd = -1;

for (let i = 0; i < lines.length; i++) {
  if (lines[i].includes('<div className="text-[14px]">') && i > 1790 && i < 1810) {
    sourcesStart = i + 1;
    // Find end - look for the </div> after all Source Items
    for (let j = i + 1; j < Math.min(i + 100, lines.length); j++) {
      if (lines[j].includes('</div>') && lines[j].trim() === '</div>' && lines[j - 1].includes(')}')) {
        sourcesEnd = j;
        break;
      }
    }
    break;
  }
}

if (sourcesStart > 0 && sourcesEnd > 0) {
  const newSources = [
    '                      <DynamicSourceItems ',
    '                        sourcesItems={sourcesItems}',
    '                        sourceContent={sourceContent}',
    '                      />'
  ];
  lines.splice(sourcesStart, sourcesEnd - sourcesStart, ...newSources);
  console.log(`✅ Fixed source items (lines ${sourcesStart+1}-${sourcesEnd+1})`);
}

// Write back
fs.writeFileSync(filePath, lines.join('\n'), 'utf8');
console.log('\n✅✅✅ DONE! File has been fixed.\n');
console.log('Now refresh your browser and test with:');
console.log('Home → "Research jurisdictional requirements"');
